### ISSUE TITLE
LXD build fails when using HCL syntax

### ISSUE THREAD

**Issue by artyomturkin** [hashicorp/packer#8932]
#### Overview of the Issue

WHen using hcl syntax with lxd source. Build fails with error

### Packer version

Packer v1.5.4

### Simplified Packer Buildfile

```hcl
source "lxd" "seed-aa" {
    image = "ubuntu-minimal:bionic"
    output_image = "cloud-seed"
    
    publish_properties {
        description = "description"
    }
}

build {
    sources = [ "source.lxd.seed-aa"]

    provisioner "shell" {
        inline = ["apt-get update"]
    }
}

```

### Operating system and Environment details

Ubuntu 19.10

### Log Fragments and crash.log files

```
==> lxd: Creating container...
2020/03/22 15:33:43 packer-builder-lxd plugin: Executing lxc command: []string{"launch", "--ephemeral=false", "--profile=default", "ubuntu-minimal:bionic", "packer-"}
2020/03/22 15:33:43 packer-builder-lxd plugin: stdout: Creating packer-
2020/03/22 15:33:43 packer-builder-lxd plugin: stderr: Error: Failed instance creation: Invalid instance name: Name must not end with "-" character
==> lxd: Error creating container: LXD command error: Error: Failed instance creation: Invalid instance name: Name must not end with "-" character
2020/03/22 15:33:43 packer-builder-lxd plugin: Executing lxc command: []string{"delete", "--force", "packer-"}
==> lxd: Unregistering and deleting deleting container...
2020/03/22 15:33:43 packer-builder-lxd plugin: stdout:
==> lxd: Error deleting container: LXD command error: Error: not found
2020/03/22 15:33:43 packer-builder-lxd plugin: stderr: Error: not found
2020/03/22 15:33:43 [INFO] (telemetry) ending 
Build 'lxd' errored: Error creating container: LXD command error: Error: Failed instance creation: Invalid instance name: Name must not end with "-" character
2020/03/22 15:33:43 machine readable: error-count []string{"1"}
```

**Comment by azr** [comment]
Hello there, thanks for opening;

Looking at the logs:

> Executing lxc command: []string{"launch", "--ephemeral=false", "--profile=default", "ubuntu-minimal:bionic", "packer-"}

...

> Name must not end with "-" character

It looks like naming things `packer-` is invalid.

https://github.com/hashicorp/packer/blob/dde74232f251434715cb76664426d9c1e1c91bd8/builder/lxd/config.go#L61-L63

In HCL `PackerBuildName` is not defaulted yet, leaving it empty. One way to fix this would be to re-enable this behaviour in packer HCL. To do this; one option could be to default it again.

**Comment by azr** [comment]
@artyomturkin, just to be sure did that work for you in the old JSON format ?

**Comment by artyomturkin** [comment]
@azr Yes. It works in the old json format

**Comment by azr** [comment]
Ah I forgot to mention this is fixed in #8956 and binaries can be found [here](https://circleci.com/gh/hashicorp/packer/44258#artifacts/containers/0). Please tell us if something is wrong.
